contracts/integrations/test/actors/staker.ts

```diff
@@ -44,6 +65,79 @@ export function StakerMixin<TBase extends Constructor>(Base: TBase): TBase & Con
                 );
             }
         }
+
+        private async *_validStake(
+            simulationEnvironment: SimulationEnvironment,
+        ): AsyncIterableIterator<AssertionResult> {
+            const { zrx } = this.actor.deployment.tokens;
+            const { deployment, balanceStore, globalStake } = simulationEnvironment;
+            const assertion = validStakeAssertion(deployment, balanceStore, globalStake, this.stake);
+
+            while (true) {
+                await simulationEnvironment.balanceStore.updateErc20BalancesAsync();
+                const zrxBalance = simulationEnvironment.balanceStore.balances.erc20[this.actor.address][zrx.address];
+                const amount = getRandomInteger(0, zrxBalance);
```

Comment:
What happens when `zrxBalance == 0`?

Reply:
since we also have `unstake` operations, this won't happen in the simulation